sql/core/src/test/resources/tpcds-plan-stability/approved-plans-v1_4/q16/explain.txt

```diff
@@ -1,235 +1,247 @@
-                        :     :              +- * ColumnarToRow (15)
-                        :     :                 +- Scan parquet default.date_dim (14)
-                        :     +- BroadcastExchange (25)
-                        :        +- * Project (24)
-                        :           +- * Filter (23)
-                        :              +- * ColumnarToRow (22)
-                        :                 +- Scan parquet default.customer_address (21)
-                        +- BroadcastExchange (32)
-                           +- * Project (31)
-                              +- * Filter (30)
-                                 +- * ColumnarToRow (29)
-                                    +- Scan parquet default.call_center (28)
+                        :     :     :  :     +- BroadcastExchange (8)
+                        :     :     :  :        +- * Project (7)
+                        :     :     :  :           +- * Filter (6)
+                        :     :     :  :              +- * ColumnarToRow (5)
+                        :     :     :  :                 +- Scan parquet default.catalog_sales (4)
+                        :     :     :  +- BroadcastExchange (14)
+                        :     :     :     +- * Filter (13)
+                        :     :     :        +- * ColumnarToRow (12)
+                        :     :     :           +- Scan parquet default.catalog_returns (11)
+                        :     :     +- BroadcastExchange (20)
+                        :     :        +- * Project (19)
+                        :     :           +- * Filter (18)
+                        :     :              +- * ColumnarToRow (17)
+                        :     :                 +- Scan parquet default.date_dim (16)
+                        :     +- BroadcastExchange (27)
+                        :        +- * Project (26)
+                        :           +- * Filter (25)
+                        :              +- * ColumnarToRow (24)
+                        :                 +- Scan parquet default.customer_address (23)
+                        +- BroadcastExchange (34)
+                           +- * Project (33)
+                              +- * Filter (32)
+                                 +- * ColumnarToRow (31)
+                                    +- Scan parquet default.call_center (30)
 
 
 (1) Scan parquet default.catalog_sales
 Output [7]: [cs_ship_date_sk#1, cs_ship_addr_sk#2, cs_call_center_sk#3, cs_warehouse_sk#4, cs_order_number#5, cs_ext_ship_cost#6, cs_net_profit#7]
 Batched: true
 Location: InMemoryFileIndex [file:/Users/yi.wu/IdeaProjects/spark/sql/core/spark-warehouse/org.apache.spark.sql.TPCDSV1_4_PlanStabilitySuite/catalog_sales]
-PushedFilters: [IsNotNull(cs_ship_date_sk), IsNotNull(cs_ship_addr_sk), IsNotNull(cs_call_center_sk)]
+PushedFilters: [IsNotNull(cs_ship_date_sk), IsNotNull(cs_ship_addr_sk), IsNotNull(cs_call_center_sk), IsNotNull(cs_order_number), IsNotNull(cs_warehouse_sk)]
```

Comment:
The inferred `IsNotNull` can filter more data.
Before this change:
![image](https://user-images.githubusercontent.com/5399861/91031650-78a4ac80-e633-11ea-8c6d-c8182dddf988.png)



After this change:
![image](https://user-images.githubusercontent.com/5399861/91031168-ccfb5c80-e632-11ea-8135-8f1e093d5f66.png)